### PR TITLE
Feature/add items to sections

### LINKED ISSUE #76: Update Wishlist item and Budget when an item is associated to a LogSection
## Expected Behavior
When we assign a wishlist item to a LogSection  hence updating its status from a `want` to a  `shopping` item we should also update the budget's status  accordingly

## Actual Behavior
This is not covered in at the moment.

## Acceptance Criteria
Adding an Item to a section should update its status and the budget's status

## Specifications

  - Version: 0.0.2-rclogs

### PR DESCRIPTION
**Pull request by zacck-zz**
Fixes #76

## Proposed Changes
  - Add changeset to associate a logsection to item 
  - Add functionality to update budget status when item is associated to a logsection

### CODE REVIEW

**Review by zacck-zz — COMMENTED** [review]
Here the work is flawed, The assumption here is that an item could only belong to one log section. However when we have items that are purchased more than once in a period (the length of a MoneyLog) then items could belong to more than one log section. In essence here we have a many to many relationship between wishlist items and log_sections

**Comment by zacck-zz** [comment]
## TODO 
Change association functionality to accommodate items belonging to more than one section,
We will need to watch out to allow items to be associated only with as many sections as the QPM amount. e.g if we have an item `soap` with a QPM amount of 3  then this item only be associated with at most 3 sections

File: lib/spender/wish_list/item.ex [hunk 1]
```diff
@@ -39,6 +39,6 @@ defmodule Spender.WishList.Item do
   def add_to_section(%Item{} = item, %LogSection{} = section) do
     item
     |> changeset(%{})
```
Comment (zacck-zz) [inline lib/spender/wish_list/item.ex]: ## Notes
Before associating an Item to a section we should run the following checks 
- associate an item with a section only if the number of associated `log_sections` is less than that of the item's `qpm`

**Review by zacck-zz — COMMENTED** [review]
worked out

**Comment by zubairshokh** [comment]
This looks perfect.

**Pull request merged**